### PR TITLE
[ci] elpi 1.10.2

### PR DESCRIPTION
**Pull request by gares**
This PR bumps elpi to 1.10.2 and the overlay moves coq-elpi/coq-master forward

*̶I̶m̶p̶o̶r̶t̶a̶n̶t̶*̶:̶ ̶t̶h̶i̶s̶ ̶P̶R̶ ̶a̶l̶s̶o̶ ̶b̶u̶m̶p̶s̶ ̶o̶c̶a̶m̶l̶ ̶t̶o̶ ̶4̶.̶0̶7̶.̶1̶ ̶a̶n̶d̶ ̶d̶u̶n̶e̶ ̶t̶o̶ ̶2̶.̶0̶.̶1̶ ̶(̶t̶h̶e̶ ̶p̶r̶e̶v̶i̶o̶u̶s̶ ̶d̶o̶c̶k̶e̶r̶ ̶f̶i̶l̶e̶s̶ ̶d̶o̶e̶s̶ ̶n̶o̶t̶ ̶w̶o̶r̶k̶ ̶a̶n̶y̶m̶o̶r̶e̶,̶ ̶s̶e̶e̶ ̶c̶o̶m̶m̶e̶n̶t̶s̶ ̶b̶e̶l̶o̶w̶)̶

### CODE REVIEW

**Comment by gares** [comment]
Dear @coq/ci-maintainers the docker build fails with message ` [ERROR] dune = 2.0.0 unmet availability conditions: false`. Help!

**Comment by gares** [comment]
According to https://opam.ocaml.org/packages/dune/dune.2.0.0/ opam 2.0.0 only works on ocaml 4.06, while we install 4.05.

How did we manage to build the docker image with opam 2.0.0 in the first place?

**Comment by gares** [comment]
Worse, dune 2.0.0 was *removed*

  https://github.com/ocaml/opam-repository/commit/e150d3bb48a4dcadb5d7837029e6a39e8455fb3a

**Review by ejgallego — COMMENTED** [review]
Yup we are in a rough spot here; but we should not bump OCaml in this PR, as that implies shifting the base OCaml we support.

Actually dune is installable with OCaml 4.02.3 , the problem in this case is `dune-configurator` which may be pulled by lablgtk3, this forces OCaml 4.06 and in newer versions up to OCaml 4.08.

This is a serious problem for us, and it is being discussed here: https://github.com/ocaml/dune/pull/3187

The easiest solution for this PR is to install a different version of dune for edge and base; dune is only needed in base for lablgtk3, in this case 1.11.4 should work.

Any of 2.2.0 / 2.3.1 should work for edge.

**Comment by ejgallego** [comment]
Actually dune 2.0.1 + dune-configurator 1.11.4 should work for base; this was the original setup and the recommended path here.

**Review by ejgallego — CHANGES_REQUESTED** [review]
- don't bump the OCaml version
- bump Dune to 2.0.1

**Comment by gares** [comment]
According to: https://opam.ocaml.org/packages/dune/dune.2.0.1/ dune 2.0.1 needs 4.06.
Anyway, I'm doing as you say, let's see

**Comment by ejgallego** [comment]
> According to: https://opam.ocaml.org/packages/dune/dune.2.0.1/ dune 2.0.1 needs 4.06.
> Anyway, I'm doing as you say, let's see

Read the constraints carefully, there is and "or" there. If OCaml is less than 4.06 then it installs the OCaml secondary compiler, so the problem does indeed come with the interface of dune <-> dune configurator but only in Dune 2.3

Note that we recommend you bump the cachekey further, if the previous image was already built the infrastructure may pick the old one. In fact, if you have two commits modifying dockerfile you should bump in each of them.

You may also may want to squash both elpi commits.

**Comment by ejgallego** [comment]
> if the previous image was already built the infrastructure may pick the old one.

Yup, indeed your build is bogus for that reason.

**Comment by gares** [comment]
Ah, I misread. It seems to work (for docker). I'll check if elpi worked tomorrow.

**Comment by ejgallego** [comment]
> Ah, I misread. It seems to work (for docker). I'll check if elpi worked tomorrow.

The docker setup is incorrect as I mentioned above, you reused the old image.

File: dev/ci/docker/bionic_coq/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ ENV COMPILER="4.05.0"
 # Common OPAM packages.
 # `num` does not have a version number as the right version to install varies
 # with the compiler version.
-ENV BASE_OPAM="num ocamlfind.1.8.1 dune.2.0.0 ounit.2.0.8 odoc.1.4.2" \
+ENV BASE_OPAM="num ocamlfind.1.8.1 dune.2.0.1 ounit.2.0.8 odoc.1.4.2" \
```
Comment (ejgallego) [inline dev/ci/docker/bionic_coq/Dockerfile]: This commit is incorrect, you change the docker image but don't bump the version.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ stages:
 variables:
   # Format: $IMAGE-V$DATE [Cache is not used as of today but kept here
   # for reference]
-  CACHEKEY: "bionic_coq-V2019-12-08-V82"
+  CACHEKEY: "bionic_coq-V2020-02-28-V83"
```
Comment (ejgallego) [inline .gitlab-ci.yml]: The CACHEKEY is incorrect [you are reusing an image, so you either bump of delete the old image in the interface]. Also, the bump is incorrect as it results in a non-buildable image due to dune problems.

**Comment by ejgallego** [comment]
I'll fix this myself as it is going to be quicker.

**Comment by gares** [comment]
thanks

**Comment by ejgallego** [comment]
Ok @gares you can update the upstream branch.

**Review by ejgallego — APPROVED** [review]

**Pull request merged**